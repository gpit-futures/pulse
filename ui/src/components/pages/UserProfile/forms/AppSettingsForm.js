import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
