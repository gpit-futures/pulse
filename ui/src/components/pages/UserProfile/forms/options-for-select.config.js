const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
