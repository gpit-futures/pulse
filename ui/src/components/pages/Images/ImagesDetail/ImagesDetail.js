import React, { PureComponent } from 'react';
import Swiper from 'react-id-swiper';
import 'react-id-swiper/src/styles/scss/swiper.scss';
import moment from 'moment';
import _ from 'lodash/fp';
import * as cornerstone from 'cornerstone-core';
import * as cornerstoneTools from 'cornerstone-tools';

import PluginDetailPanel from '../../../plugin-page-component/PluginDetailPanel';
import ImagesDetailPanel from '../images-page-component/ImagesDetailPanel';
import CornerstoneImage from '../images-page-component/CornerstoneImage';
import { valuesNames, valuesLabels } from '../forms.config';

const IMAGES_PANEL = 'imagesPanel';
const IMAGES_DETAIL_PANEL = 'imagesDetailPanel';

export default class ImagesDetail extends PureComponent {
  state = {
    touchMode: true,
    isVisibleCornerstone: true,
    isImageLoaded: true,
  };

  componentDidMount() {
    this.updateSwiper();
  }

  componentWillReceiveProps(nextProps) {
    const { isImageLoaded } = this.state;
    if ((nextProps.instanceIds[0] !== this.props.instanceIds[0]) && this.props.instanceIds.length) {
      this.setState({ touchMode: true });
      this.visibleCornerstone(false);
      this.swiper.allowTouchMove = true;
      const cornerstoneElement = this.getImgBlock();
      /* istanbul ignore next */
      if (cornerstoneElement && isImageLoaded) {
        this.disableCornerstoneTools(cornerstoneElement);
        cornerstone.reset(cornerstoneElement);
      }
    }
  }

  componentDidUpdate() {
    this.updateSwiper();
  }

  getImgBlock = () => document.getElementById(`dicomImage-${this.swiper.activeIndex}`);

  getURLtoImage = id => `${window.location.protocol}//46.101.95.245/orthanc/instances/${id}/preview`;

  imageLoaded = isImageLoaded => this.setState({ isImageLoaded });

  visibleCornerstone = isVisibleCornerstone => this.setState({ isVisibleCornerstone });

  updateSwiper = () => {
    if (this.swiper) {
      this.swiper.update();
    }
  };

  disableCornerstoneTools = (cornerstoneElement) => {
    cornerstoneTools.mouseInput.disable(cornerstoneElement);
    cornerstoneTools.pan.disable(cornerstoneElement);
    cornerstoneTools.wwwc.disable(cornerstoneElement);
  };

  zoomin = /* istanbul ignore next */ () => {
    const cornerstoneElement = this.getImgBlock();
    const viewport = cornerstone.getViewport(cornerstoneElement);
    viewport.scale += 0.1;
    cornerstone.setViewport(cornerstoneElement, viewport);
  };
  zoomout = /* istanbul ignore next */ () => {
    const cornerstoneElement = this.getImgBlock();
    const viewport = cornerstone.getViewport(cornerstoneElement);
    if (viewport.scale > 0.1) {
      viewport.scale -= 0.1;
      cornerstone.setViewport(cornerstoneElement, viewport);
    }
  };

  moveImg = /* istanbul ignore next */ () => {
    const { touchMode } = this.state;
    const cornerstoneElement = this.getImgBlock();
    if (touchMode) {
      this.setState({ touchMode: false });
      this.swiper.allowTouchMove = false;
      cornerstoneTools.mouseInput.enable(cornerstoneElement);
      cornerstoneTools.pan.activate(cornerstoneElement, 1);
      cornerstoneTools.wwwc.disable(cornerstoneElement)
    } else {
      this.setState({ touchMode: true });
      this.swiper.allowTouchMove = true;
      this.disableCornerstoneTools(cornerstoneElement)
    }
  };

  fadeImg = /* istanbul ignore next */ () => {
    const { touchMode } = this.state;
    const cornerstoneElement = this.getImgBlock();
    if (!touchMode) {
      cornerstoneTools.mouseInput.enable(cornerstoneElement);
      cornerstoneTools.mouseWheelInput.enable(cornerstoneElement);
      cornerstoneTools.pan.disable(cornerstoneElement);
      cornerstoneTools.wwwc.activate(cornerstoneElement, 2)
    } else {
      this.swiper.allowTouchMove = true;
      this.disableCornerstoneTools(cornerstoneElement);
    }
    this.setState({ touchMode: true });
  };

  render() {
    const { onExpand, onShow, openedPanel, expandedPanel, currentPanel, onEdit, editedPanel, instanceIds } = this.props;
    const { touchMode, isVisibleCornerstone } = this.state;
    let { detail } = this.props;
    detail = detail || {};
    const seriesDate = (detail[valuesNames.SERIES_DATE]) ? moment(detail[valuesNames.SERIES_DATE]).format('DD-MMM-YYYY') : '';
    const seriesTime = (detail[valuesNames.SERIES_TIME]) ? moment(detail[valuesNames.SERIES_TIME]).format('HH:MM') : '';
    const params = {
      pagination: {
        el: '.swiper-pagination',
        type: 'bullets',
        clickable: true,
      },
      navigation: {
        nextEl: '.swiper-button-next',
        prevEl: '.swiper-button-prev',
      },
      spaceBetween: 30,
      init: true,
      allowTouchMove: touchMode,
    };

    let swiperSlide;
    if (!_.isEmpty(instanceIds)) {
      swiperSlide = instanceIds.map((item, index) => {
        return (
          <div key={index}>
            <CornerstoneImage
              instanceIds={instanceIds}
              imageId={this.getURLtoImage(item)}
              visibleCornerstone={this.visibleCornerstone}
              imageLoaded={this.imageLoaded}
              isVisibleCornerstone={isVisibleCornerstone}
              index={index}
            />
          </div>
        );
      }
      )
    }

    return (
      <div className="section-detail">
        <div className="panel-group accordion">
          {(expandedPanel === IMAGES_PANEL || expandedPanel === 'all') ? <ImagesDetailPanel
            onExpand={onExpand}
            name={IMAGES_PANEL}
            title="Image(s) Series"
            isOpen={openedPanel === IMAGES_PANEL}
            currentPanel={currentPanel}
            isBtnShowPanel
            onShow={onShow}
            zoomin={this.zoomin}
            zoomout={this.zoomout}
            moveImg={this.moveImg}
            fadeImg={this.fadeImg}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper dicom-container">
                  <Swiper
                    {...params}
                    ref={node => this.swiper = node ? node.swiper : null}
                  >
                    { swiperSlide || <div /> }
                  </Swiper>
                </div>
              </div>
            </div>
          </ImagesDetailPanel> : null}
          {(expandedPanel === IMAGES_DETAIL_PANEL || expandedPanel === 'all') ? <PluginDetailPanel
            onExpand={onExpand}
            name={IMAGES_DETAIL_PANEL}
            title="Image Details"
            isOpen={openedPanel === IMAGES_DETAIL_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            isBtnShowPanel
            isShowControlPanel={false}
            onShow={onShow}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">
                  <div className="form-group">
                    <label className="control-label">{valuesLabels.MODALITY}</label>
                    <div className="form-control-static">{detail[valuesNames.MODALITY]}</div>
                  </div>
                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.PROTOCOL_NAME}</label>
                        <div className="form-control-static">{detail[valuesNames.PROTOCOL_NAME]}</div>
                      </div>
                    </div>
                    <div className="col-expand-right">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.STATION_NAME}</label>
                        <div className="form-control-static">{detail[valuesNames.STATION_NAME]}</div>
                      </div>
                    </div>
                  </div>
                  <div className="form-group">
                    <label className="control-label">{valuesLabels.OPERATORS_NAME}</label>
                    <div className="form-control-static">{detail[valuesNames.OPERATORS_NAME]}</div>
                  </div>
                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.SERIES_DATE}</label>
                        <div className="form-control-static">{seriesDate}</div>
                      </div>
                    </div>
                    <div className="col-expand-right">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.SERIES_TIME}</label>
                        <div className="form-control-static">{seriesTime}</div>
                      </div>
                    </div>
                  </div>
                  <div className="form-group">
                    <label className="control-label">{valuesLabels.AUTHOR}</label>
                    <div className="form-control-static">{detail[valuesNames.AUTHOR]}</div>
                  </div>
                  <div className="form-group">
                    <label className="control-label">{valuesLabels.SERIES_NUMBER}:</label>
                    <div className="form-control-static">{detail[valuesNames.SERIES_NUMBER]}</div>
                  </div>
                </div>
              </div>
            </div>
          </PluginDetailPanel> : null}
        </div>
      </div>
    )
  }
}
