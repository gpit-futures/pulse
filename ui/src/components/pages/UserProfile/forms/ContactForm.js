import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
