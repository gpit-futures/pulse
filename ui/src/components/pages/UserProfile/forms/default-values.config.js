import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
