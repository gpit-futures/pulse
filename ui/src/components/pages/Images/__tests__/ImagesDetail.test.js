import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import moment from 'moment';

import ImagesDetail from '../ImagesDetail/ImagesDetail';
import { valuesNames, valuesLabels } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

const propsForImagePanel = {
  detail: {
    [valuesNames.SOURCE]: 'orthanc',
    [valuesNames.SOURCE_ID]: '00d87cf2-13d39285-50905549-3faf78a7-59656dd3',
    [valuesNames.MODALITY]: 'CT',
    [valuesNames.SERIES_DATE]: 736124400000,
    [valuesNames.SERIES_TIME]: 37644000,
    [valuesNames.STATION_NAME]: 'CT01OC0',
    [valuesNames.OPERATORS_NAME]: 'Anonymized',
    [valuesNames.SERIES_NUMBER]: '365',
    [valuesNames.AUTHOR]: 'test',
    instanceIds: [
      'd1d4875a-2a8267c2-2ee4f8f0-3d9e917e-9a141930',
    ],
    [valuesNames.PROTOCOL_NAME]: 'test',
  },
  instanceIds: [
    'ecd967b0-aaec17a2-860b3925-91d8ef8b-1932d1c7',
  ],
};

const IMAGES_PANEL = 'imagesPanel';
const IMAGES_DETAIL_PANEL = 'imagesDetailPanel';
const CONVERT_SERIES_DATE = moment(propsForImagePanel.detail[valuesNames.SERIES_DATE]).format('DD-MMM-YYYY');
const CONVERT_SERIES_TIME = moment(propsForImagePanel.detail[valuesNames.SERIES_TIME]).format('HH:MM');
const EVENT = {
  preventDefault: () => {},
  removeEventListener: () => {},
  target: {
    className: 'swiper-slide',
    querySelector: () => {},
    id: 'img-0',
    style: {
      marginTop: 20,
      marginLeft: 20,
    },
  },
  clientX: 10,
  clientY: 10,
};

describe('Component <ImagesDetail />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(<ImagesDetail
      instanceIds={['ecd967b0-aaec17a2-860b3925-91d8ef8b-1932d1c7']}
    />);

    // Testing component when detail filled object, expandedPanel is all, and panel not edited
    component.setProps({ detail: propsForImagePanel.detail, expandedPanel: 'all', editedPanel: {}, instanceIds: propsForImagePanel.instanceIds });
    expect(component.props().className).toEqual('section-detail');
    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('ImagesDetailPanel')).toHaveLength(1);

    // Testing imagePanel
    expect(component.find('ImagesDetailPanel').at(0).props().name).toEqual(IMAGES_PANEL);
    expect(component.find('ImagesDetailPanel').at(0).props().title).toEqual('Image(s) Series');
    expect(component.find('ImagesDetailPanel').at(0).props().isOpen).toEqual(false);
    expect(component.find('ImagesDetailPanel').at(0).props().isBtnShowPanel).toEqual(true);

    expect(component.find('.dicom-container').at(0)).toHaveLength(1);
    expect(component.find('Swiper').at(0)).toHaveLength(1);

    // Testing imagesDetailPanel
    expect(component.find('PluginDetailPanel').at(0).props().name).toEqual(IMAGES_DETAIL_PANEL);
    expect(component.find('PluginDetailPanel').at(0).props().title).toEqual('Image Details');
    expect(component.find('PluginDetailPanel').at(0).props().isOpen).toEqual(false);
    expect(component.find('PluginDetailPanel').at(0).props().isBtnShowPanel).toEqual(true);
    expect(component.find('PluginDetailPanel').at(0).props().isShowControlPanel).toEqual(false);

    expect(component.find('.control-label').at(0).text()).toEqual(valuesLabels.MODALITY);
    expect(component.find('.control-label').at(1).text()).toEqual(valuesLabels.PROTOCOL_NAME);
    expect(component.find('.control-label').at(2).text()).toEqual(valuesLabels.STATION_NAME);
    expect(component.find('.control-label').at(3).text()).toEqual(valuesLabels.OPERATORS_NAME);
    expect(component.find('.control-label').at(4).text()).toEqual(valuesLabels.SERIES_DATE);
    expect(component.find('.control-label').at(5).text()).toEqual(valuesLabels.SERIES_TIME);
    expect(component.find('.control-label').at(6).text()).toEqual(valuesLabels.AUTHOR);
    expect(component.find('.control-label').at(7).text()).toEqual(`${valuesLabels.SERIES_NUMBER}:`);

    expect(component.find('.form-control-static').at(0).text()).toEqual(propsForImagePanel.detail[valuesNames.MODALITY]);
    expect(component.find('.form-control-static').at(1).text()).toEqual(propsForImagePanel.detail[valuesNames.PROTOCOL_NAME]);
    expect(component.find('.form-control-static').at(2).text()).toEqual(propsForImagePanel.detail[valuesNames.STATION_NAME]);
    expect(component.find('.form-control-static').at(3).text()).toEqual(propsForImagePanel.detail[valuesNames.OPERATORS_NAME]);
    expect(component.find('.form-control-static').at(4).text()).toEqual(CONVERT_SERIES_DATE);
    expect(component.find('.form-control-static').at(5).text()).toEqual(CONVERT_SERIES_TIME);
    expect(component.find('.form-control-static').at(6).text()).toEqual(propsForImagePanel.detail[valuesNames.AUTHOR]);
    expect(component.find('.form-control-static').at(7).text()).toEqual(propsForImagePanel.detail[valuesNames.SERIES_NUMBER]);

    expect(component).toMatchSnapshot();

    // Testing methods shallow

    expect(component.state().isVisibleCornerstone).toEqual(true);
    component.instance().visibleCornerstone(false);
    expect(component.state().isVisibleCornerstone).toEqual(false);
    component.instance().visibleCornerstone(true);
    expect(component.state().isVisibleCornerstone).toEqual(true);
    component.instance().disableCornerstoneTools(EVENT);

  });

  it('should renders with props correctly mount', () => {
    const component = mount(<ImagesDetail
      instanceIds={['ecd967b0-aaec17a2-860b3925-91d8ef8b-1932d1c7']}
      detail={propsForImagePanel.detail}
      expandedPanel="all"
    />);

    // Testing methods

    component.instance().getImgBlock();
    component.instance().getURLtoImage('test');

    component.setProps({ instanceIds: ['ecd967b0-aaec17a2-860b3925-91d8ef8b-1932d1c9'] });
    // expect(component).toMatchSnapshot();

    component.setProps({ instanceIds: [] });
    // expect(component).toMatchSnapshot();
  });

  it('should renders correctly with different state of props', () => {
    const component = shallow(
      <ImagesDetail
        instanceIds={['ecd967b0-aaec17a2-860b3925-91d8ef8b-1932d1c8']}
      />);
    // // Testing component when detail empty object, expandedPanel is imagePanel
    component.setProps({ detail: { [valuesNames.SERIES_DATE]: 1507020019000, [valuesNames.SERIES_TIME]: 37644000 }, expandedPanel: IMAGES_PANEL, editedPanel: {} });
    expect(component.find('ImagesDetailPanel')).toHaveLength(1);
    expect(component.find('ImagesDetailPanel').props().name).toEqual(IMAGES_PANEL);
    expect(component).toMatchSnapshot();

    // Testing component when detail empty object, expandedPanel is imagesDetailPanel
    component.setProps({ detail: {}, expandedPanel: IMAGES_DETAIL_PANEL });
    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(IMAGES_DETAIL_PANEL);
    expect(component).toMatchSnapshot();
  });
});

