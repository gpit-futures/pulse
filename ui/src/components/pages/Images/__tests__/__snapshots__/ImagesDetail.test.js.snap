// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ImagesDetail /> should renders correctly with different state of props 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <ImagesDetailPanel
      fadeImg={[Function]}
      isBtnShowPanel={true}
      isOpen={false}
      moveImg={[Function]}
      name="imagesPanel"
      title="Image(s) Series"
      zoomin={[Function]}
      zoomout={[Function]}
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper dicom-container"
          >
            <ReactIdSwiper
              ContainerEl="div"
              WrapperEl="div"
              allowTouchMove={true}
              containerClass="swiper-container"
              init={true}
              navigation={
                Object {
                  "nextEl": ".swiper-button-next",
                  "prevEl": ".swiper-button-prev",
                }
              }
              pagination={
                Object {
                  "clickable": true,
                  "el": ".swiper-pagination",
                  "type": "bullets",
                }
              }
              slideClass="swiper-slide"
              spaceBetween={30}
              wrapperClass="swiper-wrapper"
            >
              <div
                key="0"
              >
                <CornerstoneImage
                  imageId="http://46.101.95.245/orthanc/instances/ecd967b0-aaec17a2-860b3925-91d8ef8b-1932d1c8/preview"
                  imageLoaded={[Function]}
                  index={0}
                  instanceIds={
                    Array [
                      "ecd967b0-aaec17a2-860b3925-91d8ef8b-1932d1c8",
                    ]
                  }
                  isVisibleCornerstone={true}
                  visibleCornerstone={[Function]}
                />
              </div>
            </ReactIdSwiper>
          </div>
        </div>
      </div>
    </ImagesDetailPanel>
  </div>
</div>
`;

exports[`Component <ImagesDetail /> should renders correctly with different state of props 2`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={Object {}}
      isBtnShowPanel={true}
      isOpen={false}
      isShowControlPanel={false}
      name="imagesDetailPanel"
      title="Image Details"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Modality
              </label>
              <div
                className="form-control-static"
              />
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Protocol Name
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Station Name
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Operator Name
              </label>
              <div
                className="form-control-static"
              />
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Series Date
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Series Time
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Author
              </label>
              <div
                className="form-control-static"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Series No:
                :
              </label>
              <div
                className="form-control-static"
              />
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <ImagesDetail /> should renders with props correctly 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <ImagesDetailPanel
      fadeImg={[Function]}
      isBtnShowPanel={true}
      isOpen={false}
      moveImg={[Function]}
      name="imagesPanel"
      title="Image(s) Series"
      zoomin={[Function]}
      zoomout={[Function]}
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper dicom-container"
          >
            <ReactIdSwiper
              ContainerEl="div"
              WrapperEl="div"
              allowTouchMove={true}
              containerClass="swiper-container"
              init={true}
              navigation={
                Object {
                  "nextEl": ".swiper-button-next",
                  "prevEl": ".swiper-button-prev",
                }
              }
              pagination={
                Object {
                  "clickable": true,
                  "el": ".swiper-pagination",
                  "type": "bullets",
                }
              }
              slideClass="swiper-slide"
              spaceBetween={30}
              wrapperClass="swiper-wrapper"
            >
              <div
                key="0"
              >
                <CornerstoneImage
                  imageId="http://46.101.95.245/orthanc/instances/ecd967b0-aaec17a2-860b3925-91d8ef8b-1932d1c7/preview"
                  imageLoaded={[Function]}
                  index={0}
                  instanceIds={
                    Array [
                      "ecd967b0-aaec17a2-860b3925-91d8ef8b-1932d1c7",
                    ]
                  }
                  isVisibleCornerstone={true}
                  visibleCornerstone={[Function]}
                />
              </div>
            </ReactIdSwiper>
          </div>
        </div>
      </div>
    </ImagesDetailPanel>
    <PluginDetailPanel
      editedPanel={Object {}}
      isBtnShowPanel={true}
      isOpen={false}
      isShowControlPanel={false}
      name="imagesDetailPanel"
      title="Image Details"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Modality
              </label>
              <div
                className="form-control-static"
              >
                CT
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Protocol Name
                  </label>
                  <div
                    className="form-control-static"
                  >
                    test
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Station Name
                  </label>
                  <div
                    className="form-control-static"
                  >
                    CT01OC0
                  </div>
                </div>
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Operator Name
              </label>
              <div
                className="form-control-static"
              >
                Anonymized
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Series Date
                  </label>
                  <div
                    className="form-control-static"
                  >
                    30-Apr-1993
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Series Time
                  </label>
                  <div
                    className="form-control-static"
                  >
                    12:01
                  </div>
                </div>
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Author
              </label>
              <div
                className="form-control-static"
              >
                test
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Series No:
                :
              </label>
              <div
                className="form-control-static"
              >
                365
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;
